Detectors/TPC/workflow/src/ZSSpec.cxx

```diff
@@ -261,34 +261,23 @@ DataProcessorSpec getZStoDigitsSpec(std::vector<int> const& tpcSectors)
         o2::framework::RawParser parser(raw.data(), raw.size());
 
         const unsigned char* ptr = nullptr;
-        int count = 0;
         int cruID = 0;
-        rdh_utils::FEEIDType lastFEE = -1;
+        rdh_utils::FEEIDType FEEID = -1;
         size_t totalSize = 0;
         for (auto it = parser.begin(); it != parser.end(); it++) {
           const unsigned char* current = it.raw();
           const o2::header::RAWDataHeader* rdh = (const o2::header::RAWDataHeader*)current;
-          if (current == nullptr || it.size() == 0 || (current - ptr) % TPCZSHDR::TPC_ZS_PAGE_SIZE || o2::raw::RDHUtils::getFEEID(*rdh) != lastFEE) {
-            if (count) {
-              unsigned int sector = cruID / 10;
-              gsl::span<const ZeroSuppressedContainer8kb> z0in(reinterpret_cast<const ZeroSuppressedContainer8kb*>(ptr), count);
-              decoder->DecodeZSPages(&z0in, &outDigits[sector], firstOrbit);
-            }
-            count = 0;
-            lastFEE = o2::raw::RDHUtils::getFEEID(*rdh);
-            cruID = int(o2::raw::RDHUtils::getCRUID(*rdh));
-            if (it.size() == 0) {
-              ptr = nullptr;
-              continue;
-            }
+          if (it.size() == 0) {
```

Comment:
This seems to change the logic, are we sure it is correct?

Reply:
> Generally I am fine with it, just one comment on which @JohannesLb could comment. It should certainly be squashed.

Looks good from my point of view. This was the fix I suggested when the workflow is run with zs raw data input and digit as output of the workflow, which was broken beforehand / producing empty output.